GTViewController/Classes/Custom Classes/UISnuffleButton.m

```diff
@@ -445,15 +454,19 @@ - (void)touchesEnded:(NSSet *)touches withEvent:(UIEvent *)event {
 				
 			}
 			
-            [self.tapEvents enumerateObjectsUsingBlock:^(NSString *eventName, NSUInteger idx, BOOL *stop) {
-                if (eventName) {
-                    NSString *trimmedEventName = [eventName stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
-                    
-                    [[NSNotificationCenter defaultCenter] postNotificationName:UISnuffleButtonNotificationButtonTapEvent
-                                                                        object:self
-                                                                      userInfo:@{UISnuffleButtonNotificationButtonTapEventKeyEventName: trimmedEventName}];
-                }
-            }];
+            if (self.forceValidation && self.tapDelegate && [self.tapDelegate respondsToSelector:@selector(didReceiveTapOnPositiveButton:)]) {
```

Comment:
What happens if content creators define a positive button outside of a followup modal and add tap events to it? The way I read it nothing happens, which is a bug. You are right that it simplifies things but at the cost of having a generalized solution. I think it would be better to change it so that it makes sense to content creators, should they try something different.